core/src/main/java/org/infinispan/xsite/commands/remote/XSiteStatePushRequest.java

```diff
@@ -1,31 +1,31 @@
 package org.infinispan.xsite.commands.remote;
 
-import org.infinispan.commons.marshall.MarshallUtil;
+import java.util.concurrent.CompletionStage;
+
+import org.infinispan.commons.marshall.ProtoStreamTypeIds;
 import org.infinispan.factories.ComponentRegistry;
+import org.infinispan.protostream.annotations.ProtoFactory;
+import org.infinispan.protostream.annotations.ProtoField;
+import org.infinispan.protostream.annotations.ProtoTypeId;
 import org.infinispan.util.ByteString;
 import org.infinispan.xsite.statetransfer.XSiteState;
 
-import java.io.IOException;
-import java.io.ObjectInput;
-import java.io.ObjectOutput;
-import java.util.concurrent.CompletionStage;
-
 /**
  * Wraps the state to be sent to another site.
  * <p>
  * It used when the backup strategy is set to {@link org.infinispan.configuration.cache.BackupConfiguration.BackupStrategy#SYNC}.
  *
  * @since 15.0
  */
+@ProtoTypeId(ProtoStreamTypeIds.XSITE_STATE_PUSH_REQUEST)
 public class XSiteStatePushRequest extends XSiteCacheRequest<Void> {
 
-   private XSiteState[] chunk;
-   private long timeoutMillis;
-
-   public XSiteStatePushRequest() {
-      super(null);
-   }
+   @ProtoField(2)
+   final XSiteState[] chunk;
```

Comment:
if you feel that it is worth it, change to a List. Reading arrays is not very good with protostream; but state transfer should be rare.

Reply:
I agree we should change it, but let's do it in a subsequent PR as it impacts a few interfaces. I have created https://github.com/infinispan/infinispan/issues/14109 and added it to this epic.